### PR TITLE
[InstallAppleProvisioningProfileV1] Migrated to Node10

### PR DESCRIPTION
**Pull request by egor-bryzgalov**
**Task name**: InstallAppleProvisioningProfileV1

**Description**: Updated dependencies to prepare the task to be executed with Node 10.
[This canary test](https://github.com/microsoft/azure-pipelines-canary/pull/433) is required before merging.

**Documentation changes required:** No

**Added unit tests:** No

**Attached related issue:** [#299](https://github.com/microsoft/build-task-team/issues/299)

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

### CODE REVIEW

File: Tasks/InstallAppleProvisioningProfileV1/Tests/L0.ts [hunk 1]
```diff
@@ -11,7 +11,7 @@ describe('InstallAppleProvisioningProfile Suite', function () {
     after(() => {
     });
 
-    it('Defaults: install from SecureFile', (done: MochaDone) => {
+    it('Defaults: install from SecureFile', (done: any) => {
```
Comment (alex-peck) [inline Tasks/InstallAppleProvisioningProfileV1/Tests/L0.ts:14]: These can be changed to `Done` imported from `mocha`

Comment (egor-bryzgalov) [reply Tasks/InstallAppleProvisioningProfileV1/Tests/L0.ts]: @alex-peck Thank you! I've changed this.

File: Tasks/InstallAppleProvisioningProfileV1/package.json [hunk 2]
```diff
@@ -17,9 +17,13 @@
...
-    "@types/node": "^6.0.0",
-    "azure-pipelines-task-lib": "^2.8.0",
-    "azure-pipelines-tasks-securefiles-common": "1.0.0",
-    "azure-pipelines-tasks-ios-signing-common": "1.0.0"
+    "azure-pipelines-tasks-ios-signing-common": "2.0.0-preview",
+    "azure-pipelines-tasks-securefiles-common": "2.0.0-preview",
+    "azure-pipelines-task-lib": "^3.0.1-preview"
+  },
+  "devDependencies": {
+    "@types/mocha": "^8.0.3",
```
Comment (alex-peck) [inline Tasks/InstallAppleProvisioningProfileV1/package.json:25]: For now, move these @types dependencies from `devDependencies` to `dependencies`

There is a bug in make-build.js that is causing us not to error in this case. There is work to be done to filter out `devDependencies` from the final packaged task.

Comment (alex-peck) [reply Tasks/InstallAppleProvisioningProfileV1/package.json:25]: PR: #13731

Comment (egor-bryzgalov) [reply Tasks/InstallAppleProvisioningProfileV1/package.json]: Thank you, @alex-peck I moved them.

**Review by DaniilShmelev — APPROVED** [review]

**Review by anatolybolshakov — APPROVED** [review]

**Review by EzzhevNikita — APPROVED** [review]

**Review by alex-peck — APPROVED** [review]
Looks great, thanks!

**Review by alexander-smolyakov — APPROVED** [review]

**Review by EzzhevNikita — APPROVED** [review]

**Review by alex-peck — APPROVED** [review]

**Review by sdobrodeev — APPROVED** [review]

**Pull request merged**